cmd/check.go

```diff
@@ -437,44 +426,66 @@ func exportControlResults(ctx context.Context, executionTree *controlexecute.Exe
 }
 
 func getExportTargets(executing string) ([]controldisplay.CheckExportTarget, error) {
-	formats := []controldisplay.CheckExportTarget{}
+	targets := []controldisplay.CheckExportTarget{}
+	targetErrors := []error{}
+
 	exports := viper.GetStringSlice(constants.ArgExport)
 	for _, export := range exports {
-		var targetError error
+		export = strings.TrimSpace(export)
 
-		if len(strings.TrimSpace(export)) == 0 {
+		if len(export) == 0 {
 			// if this is an empty string, ignore
 			continue
 		}
 
-		parts := strings.SplitN(export, ":", 2)
-
-		var format, fileName string
-
-		if len(parts) == 2 {
-			// we have two distinct parts - life is good
-			format = parts[0]
-			fileName = parts[1]
-			fileName, targetError = helpers.Tildefy(fileName)
-		} else {
-			format = parts[0]
-
-			// try to get an export formatter
-			if formatter, fmtError := controldisplay.GetExportFormatter(format); fmtError != nil {
-				// this is not a valid format. assume it is a file name
-				fileName = format
-				// now infer the format from the file name
-				format, targetError = controldisplay.InferFormatFromExportFileName(fileName)
-			} else {
-				// the format was valid, generate default filename
-				fileName = generateDefaultExportFileName(formatter, executing)
+		var fileName string
+		var formatter controldisplay.Formatter
+
+		formatter, fileName, err := parseExportArg(export)
+		if err != nil {
+			targetErrors = append(targetErrors, err)
+			continue
+		}
+		if formatter == nil {
+			targetErrors = append(targetErrors, controldisplay.ErrFormatterNotFound)
+			continue
+		}
+
+		if len(fileName) == 0 {
+			fileName = generateDefaultExportFileName(formatter, executing)
+		}
+
+		newTarget := controldisplay.NewCheckExportTarget(formatter, fileName)
+		isAlreadyAdded := false
+		for _, t := range targets {
+			if t.File == newTarget.File {
+				isAlreadyAdded = true
+				break
 			}
 		}
-		formats = append(formats, controldisplay.NewCheckExportTarget(format, fileName, targetError))
+
+		if !isAlreadyAdded {
+			targets = append(targets, newTarget)
+		}
 	}
-	err := validateExportTargets(formats)
 
-	return formats, err
+	return targets, utils.CombineErrors(targetErrors...)
+}
+
+func parseExportArg(arg string) (formatter controldisplay.Formatter, targetFileName string, err error) {
+	var found bool
+	if formatter, found = controldisplay.GetDefinedExportFormatter(arg); found {
```

Comment:
we need some comments here

Reply:
also don't like the function name `GetDefinedExportFormatter` and `GetDefinedOutputFormatter`

Reply:
`GetDefinedExportFormatter` will go away when we move `CSV` and `JSON` to templates and so will `GetDefinedOutputFormatter` . 

We will probably move to something similar to:
```
if arg == "text" || arg == "brief" {
    return TableFormatter{}
} else {
    return controldisplay.GetTemplateExportFormatter(arg, true)
}
```